internal/pkg/api/handleEnroll.go

```diff
@@ -538,3 +711,56 @@ func validateRequest(ctx context.Context, data io.Reader) (*EnrollRequest, error
 
 	return &req, nil
 }
+
+func compareHashAndToken(zlog zerolog.Logger, hash string, token string, cfg config.PBKDF2) (bool, error) {
+	// format of stored replace_token
+	// $pbkdf2-sha512${iterations}${salt]${encoded}
+	// ${salt} and ${encoded} are stored base64 encoded
+	tokens := strings.Split(hash, "$")
+	if len(tokens) != 5 || tokens[0] != "" {
+		// stored hash is invalid
+		zlog.Error().Err(ErrAgentCorrupted).Msg("replace_token hash is corrupted")
+		return false, ErrAgentCorrupted
+	}
+	if tokens[1] != "pbkdf2-sha512" {
+		// unsupported hash
+		zlog.Error().Err(ErrAgentCorrupted).Msg("replace_token hash is not pbkdf2-sha512")
+		return false, ErrAgentCorrupted
+	}
+	iterations, err := strconv.Atoi(tokens[2])
+	if err != nil {
+		// hash invalid format
+		zlog.Error().Err(err).Msg("replace_token hash iterations not an integer")
+		return false, ErrAgentCorrupted
+	}
+	salt, err := base64.RawStdEncoding.DecodeString(tokens[3])
+	if err != nil {
+		// hash invalid format
+		zlog.Error().Err(err).Msg("replace_token hash failed to base64 decode salt")
+		return false, ErrAgentCorrupted
+	}
+	encoded, err := base64.RawStdEncoding.DecodeString(tokens[4])
+	if err != nil {
+		// hash invalid format
+		zlog.Error().Err(err).Msg("replace_token hash failed to base64 decode encoded")
+		return false, ErrAgentCorrupted
+	}
+	key := pbkdf2.Key([]byte(token), salt, iterations, cfg.KeyLength, sha512.New)
+	return bytes.Equal(key, encoded), nil
```

Comment:
```suggestion
	return hmac.Equal(key, encoded), nil
```

Reply:
Why `hmac.Equal`? Whats the reason? Would like to add that in the code comment so its clear. Its not clear to my why that is used here.

Reply:
Got it, thanks for the details. I have updated it to use `hmac.Equal` and added a comment.